book/java-rebuilds-service.md

```diff
@@ -0,0 +1,154 @@
+# JVM Build Service
+
+## Overview
+
+The JVM Build Service (JBS) is a controller that will rebuild Java and other JVM Language based dependencies from source.
+
+Java uses a binary distribution model, so the only way to ensure an application is completly built from source is to rebuild all its component libraries from source themselves. Unfortunatly due to the nature of the Java ecosystem this is not an easy process and would normally require a large amount of manual toil.
+
+JBS will attempt to automate this proccess as much as possible, making it much less time consuming to create fully supply chain controlled Java builds.
+
+### Dependencies
+
+JBS depends on Tekton, but is otherwise largely independent of the rest of RHTAP. Given the right configuration it can be deployed outside of RHTAP, which is mainly used for testing and development.
+
+## Architecture
+
+### Flow Overview
+
+The JVM Build service can be enabled on a specific namespace by creating a `JBSConfig` object, with `enableRebuilds: true` specified. This will trigger the controller to create a deployment of the local Cache in the namespace. This cache has a number of purposes, but the primary one is to cache artifacts from Maven central to speed up Java builds.
+
+Once it is enabled and the cache has started the JVM Build Service will watch for `PipelineRun` objects with the `JAVA_COMMUNITY_DEPENDENCIES` results. This will be present on end user builds that have dependencies from untrusted upstream sources.
```

Comment:
You introduce the concept of untrusted upstream sources here but you do not go into further detail. Can you include/expound upon this term in the Overview section?

Reply:
I have added a little bit. Do we want both this and the end user docs to be standalone documents? I was thinking we should just make sure all this is explained really well in the end user docs, and then explicitly point out that this doc assumes you have read the other one.

Reply:
I think as much as reasonable, it makes sense to have each document live on its own. This separation can allow you to be more technical here if desired while not going into as much detail on the user documentation.

If you think linking to the end user documentation is sufficient then I'll leave that up to you!